components/browser/session/src/main/java/mozilla/components/browser/session/Session.kt

```diff
@@ -88,6 +90,13 @@ class Session(
         _, old, new -> notifyObservers (old, new, { onSecurityChanged() })
     }
 
+    /**
+     * Configuration data in case this session is used for a Custom Tab.
+     */
+    var customTabConfig: CustomTabConfig? by Delegates.observable<CustomTabConfig?>(null) {
+        _, _, _ -> notifyObservers ({ onCustomTabConfigChanged() })
+    }
+
```

Comment:
In theory this doesn't need to be observable because it does never change for a session.

Reply:
Hm right, but won't it at least go from `null` to `not-null`. Seems that should be observable then? Not sure, but feels wrong having this one property not being observable :)

Reply:
I think in Focus we pass this in with the constructor (or should at least) because we already have this at the creation of the session (Intent). But let's see how this works once we add the intent handling.

In Focus we have a mix of observable and non-observable properties. But I agree that we may want to have all properties observable because we do not know what future apps will need. Eventually we may want to do the sames changes as in SessionManager (#294): Empty default implementations so that not all methods of the interface need to be implemented.